src/shared/ui/inputPrompter.ts

```diff
@@ -85,18 +86,21 @@ export class InputBoxPrompter extends Prompter<string> {
      * attempts to submit their response.
      *
      * @param validate Validator function
+     * @param isFinalInput true if the value was the final input (onDidAccept), false if this input is being typed by the user. For expensive validation (e.g. performs a service/SDK call) the validator can use this to skip validation until the final input.
      */
-    public setValidation(validate: (value: string) => string | undefined): void {
+    public setValidation(
+        validate: (value: string, isFinalInput: boolean) => string | undefined | Thenable<string | undefined>
+    ): void {
         this.validateEvents.forEach(d => d.dispose())
         this.validateEvents = []
 
         this.inputBox.onDidChangeValue(
-            value => (this.inputBox.validationMessage = validate(value)),
+            async value => (this.inputBox.validationMessage = await validate(value, false)),
             undefined,
             this.validateEvents
         )
         this.inputBox.onDidAccept(
-            () => (this.inputBox.validationMessage = validate(this.inputBox.value)),
+            async () => (this.inputBox.validationMessage = await validate(this.inputBox.value, true)),
```

Comment:
seems like vscode is not awaiting this.

Reply:
Ahh ok so for input boxes VSC doesn't handle async validation callbacks. We could make a contribution to their codebase for this, otherwise the solution on our end is messy.

Reply:
ok i'll try https://github.com/aws/aws-toolkit-vscode/blob/b4695d472e7d9cc5285f27d9a1bd9c10f5e88a2e/src/credentials/wizards/createProfile.ts#L69-L104 then.